plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -1021,6 +1048,7 @@ private static <K, V> LoadingCache<K, V> buildCache(
         if (statsRecording == StatsRecording.ENABLED) {
             cacheBuilder.recordStats();
         }
+        cacheBuilder.shareNothingWhenDisabled();
```

Comment:
Why do we need to invoke this now ?

Reply:
Because we use EvictableCache and as of now configure some of them not to cache ("disable")